### PR TITLE
Add `get_spot_price_after_closing_short` rust function

### PR DESCRIPTION
**Pull request by sentilesdal**
Adding this so we can add an error check to protect simulations and fuzz tests

- rename spot price functions to be long/short specfic
- add `get_spot_price_after_closing_short`
- add `get_close_short_max_spot_price`
- add negative interest error checking to `calculate_close_short`

### CODE REVIEW

File: crates/hyperdrive-math/src/short/close.rs [hunk 1]
```diff
@@ -179,4 +238,78 @@ mod tests {
...
+                .client(chain.chain().accounts()[0].clone())
+                .await?;
+            let mock = MockHyperdrive::deploy(client, (state.clone().config,))?
+                .send()
+                .await?;
+            let maturity_time = rng.gen_range(fixed!(0)..=state.position_duration());
+            let in_ = rng.gen_range(fixed!(0)..=state.bond_reserves());
+            let normalized_time_remaining = rng.gen_range(fixed!(0)..=fixed!(1e18));
+            let open_vault_share_price = rng.gen_range(fixed!(5e17)..=fixed!(10e18));
+            let close_vault_share_price = rng.gen_range(fixed!(5e17)..=fixed!(10e18));
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply crates/hyperdrive-math/src/short/close.rs:262]: They're not that big are they?  0.5e18 to 10e18

File: crates/hyperdrive-math/src/short/close.rs [hunk 2]
```diff
@@ -179,4 +238,78 @@ mod tests {
...
+        let mut rng = thread_rng();
+        for _ in 0..*FAST_FUZZ_RUNS {
+            let state = rng.gen::<State>();
+            let client = chain
+                .chain()
+                .client(chain.chain().accounts()[0].clone())
+                .await?;
+            let mock = MockHyperdrive::deploy(client, (state.clone().config,))?
+                .send()
+                .await?;
```
[the remark this answers is not in the document]
Comment (sentilesdal) [reply crates/hyperdrive-math/src/short/close.rs:257]: that's a good question, maybe not but its the easiest way to make sure we have the same starting state for each round.

File: crates/hyperdrive-math/src/short/close.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryangoree) [reply crates/hyperdrive-math/src/short/close.rs:262]: Ooh I just noticed the `17` was not `18`.

[the remark this answers is not in the document]
Comment (jalextowle) [reply crates/hyperdrive-math/src/short/close.rs:262]: Let’s actually make it 0.5e18. This is consistent with our style elsewhere.

**Pull request closed without merging**